onnxruntime/core/providers/cpu/sequence/concat_from_sequence.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cpu/sequence/concat_from_sequence.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/providers/cpu/tensor/utils.h"
+#include "core/framework/TensorSeq.h"
+
+using namespace onnxruntime::common;
+
+namespace onnxruntime {
+
+ONNX_CPU_OPERATOR_KERNEL(
+    ConcatFromSequence,
+    11,
+    KernelDefBuilder()
+        .TypeConstraint("S", DataTypeImpl::AllSequenceTensorTypes()),
+    ConcatFromSequence);
+
+// core Compute() method for the 'ConcatFromSequence' kernel
+Status ConcatFromSequence::Compute(OpKernelContext* ctx) const {
+  const auto* X = ctx->Input<TensorSeq>(0);
+  ORT_ENFORCE(X != nullptr, "Got nullptr for sequence input.");
+
+  // number of input tensors in the Sequence to concatenate
+  int input_count = static_cast<int>(X->tensors.size());
+
+  // Hold pointers to the input tensors to be used in the PrepareForCompute() step
+  const auto& input_tensors = X->tensors;
+
+  std::vector<const Tensor*> input_tensor_pointers;
```

Comment:
Why do we need this separate vector when we already have a vector of Tensor in the TensorSeq struct?

Reply:
I refactored the method in the base class so that it is re-usable for both `Concat` and `ConcatFromSequence` and hence I settled on a middle-ground base class method interface. In `Concat`, we need to use OpKernelContext to parse the inputs and in `ConcatFromSequence`, we get a vector of tensors, and so I thought we could pass in pointers to the input Tensors for both ops. This pointer accumulation should be cheap. Please let me know if you have any concerns.

Reply:
Alright, let's go with this. We can improve later if desired.